cmd/geth/chaincmd.go

```diff
@@ -268,6 +285,181 @@ func exportChain(ctx *cli.Context) error {
+				if lp.hc.GetBlockHashesFromHash(header.Hash(), uint64(skip+1))[skip] == hash {
+					hash = header.Hash()
+				} else {
+					unknown = true
+				}
+			} else {
+				unknown = true
+			}
+		}
+	}
+
+	lp.dl.DeliverHeaders("local", headers)
+	return nil
+}
+
+func (lp *localPeer) RequestHeadersByNumber(num uint64, amount int, skip int, reverse bool) error {
+	var (
+		headers []*types.Header
+		unknown bool
+	)
+
+	for !unknown && len(headers) < amount {
+		origin := lp.hc.GetHeaderByNumber(num)
+		if origin == nil {
+			break
+		}
+
+		if reverse {
+			if num >= uint64(skip+1) {
+				num -= uint64(skip + 1)
+			} else {
+				unknown = true
+			}
+		} else {
+			num += uint64(skip + 1)
+		}
+		headers = append(headers, origin)
+	}
+
+	lp.dl.DeliverHeaders("local", headers)
+	return nil
+}
+
+func (lp *localPeer) RequestBodies(hashes []common.Hash) error {
+	var (
+		transactions [][]*types.Transaction
+		uncles       [][]*types.Header
+	)
+
+	for _, hash := range hashes {
+		block := core.GetBlock(lp.chainDb, hash, lp.hc.GetBlockNumber(hash))
+		transactions = append(transactions, block.Transactions())
+		uncles = append(uncles, block.Uncles())
+	}
+
+	lp.dl.DeliverBodies("local", transactions, uncles)
+	return nil
+}
+
+func (lp *localPeer) RequestReceipts(hashes []common.Hash) error {
+	var receipts [][]*types.Receipt
+
+	for _, hash := range hashes {
+		receipts = append(receipts, core.GetBlockReceipts(lp.chainDb, hash, lp.hc.GetBlockNumber(hash)))
+	}
+
+	lp.dl.DeliverReceipts("local", receipts)
+	return nil
+}
+
+func (lp *localPeer) RequestNodeData(hashes []common.Hash) error {
+	var data [][]byte
+
+	for _, hash := range hashes {
+		if entry, err := lp.chainDb.Get(hash.Bytes()); err == nil {
+			data = append(data, entry)
+		}
+	}
+
+	lp.dl.DeliverNodeData("local", data)
+	return nil
+}
+
+func copyDb(ctx *cli.Context) error {
+	if len(ctx.Args()) != 1 {
+		utils.Fatalf("This command requires an argument.")
+	}
+
+	stack := makeFullNode(ctx)
+	chain, chainDb, mux := utils.MakeChain(ctx, stack)
+	start := time.Now()
+
+	syncmode := *utils.GlobalTextMarshaler(ctx, utils.SyncModeFlag.Name).(*downloader.SyncMode)
+	dl := downloader.New(syncmode, chainDb, mux, chain, nil, nil)
+
+	var err error
+	filename := ctx.Args().First()
+	cache := ctx.GlobalInt(utils.CacheFlag.Name)
+	handles := 256
+	localdb, err := ethdb.NewLDBDatabase(filename, cache, handles)
+	if err != nil {
+		return err
+	}
+
+	hc, err := core.NewHeaderChain(localdb, chain.Config(), chain.Engine(), func() bool { return false })
+	if err != nil {
+		return err
+	}
+
+	peer := &localPeer{localdb, hc, dl}
+	if err := dl.RegisterPeer("local", 63, peer); err != nil {
+		return err
+	}
+
+	currentHeader := hc.CurrentHeader()
+	if err := dl.Synchronise("local", currentHeader.Hash(), hc.GetTd(currentHeader.Hash(), currentHeader.Number.Uint64()), syncmode); err != nil {
+		return err
+	}
+	for dl.Synchronising() {
+		time.Sleep(1000)
```

Comment:
That's 1000ns. Please add a unit, e.g. `1 * time.Microsecond`. Maybe it should sleep longer, too.